docs: adding opa v1 docs and tests (#3908)
open-policy-agent/gatekeeper: website/docs/constrainttemplates.md

$ kubectl create ns foobar
Error from server ([ns-must-have-gk] you must provide labels: {"gatekeeper"}): admission webhook "validation.gatekeeper.sh" denied the request: [ns-must-have-gk] you must provide labels: {"gatekeeper"}
```

## Enable OPA Rego v1 syntax in ConstraintTemplates

Gatekeeper 3.19 ships with ability to use OPA Rego v1 as policy language in ConstraintTemplates. Using Rego v1 syntax is opt-in, by default only Rego v0 is allowed. You can use below spec to enable Rego v1 syntax:

```yaml
...
  targets:
    - target: admission.k8s.gatekeeper.sh
      code:
        - engine: Rego
          source:
            version: "v1"
            rego: |
              <v1-rego-code>
...
```

:::note

> Rego v1 syntax can only be used under `targets[_].code[_].[engine: Rego].source`.

Reviewer: ```suggestion
Rego v1 syntax can only be used under `targets[_].code[_].[engine: Rego].source` with `version: "v1"`
```

$ kubectl create ns foobar
Error from server ([ns-must-have-gk] you must provide labels: {"gatekeeper"}): admission webhook "validation.gatekeeper.sh" denied the request: [ns-must-have-gk] you must provide labels: {"gatekeeper"}
```

## Enable OPA Rego v1 syntax in ConstraintTemplates

Gatekeeper 3.19 ships with ability to use OPA Rego v1 as policy language in ConstraintTemplates. Using Rego v1 syntax is opt-in, by default only Rego v0 is allowed. You can use below spec to enable Rego v1 syntax:

```yaml
...
  targets:
    - target: admission.k8s.gatekeeper.sh
      code:
        - engine: Rego
          source:
            version: "v1"
            rego: |
              <v1-rego-code>
...
```

:::note

> Rego v1 syntax can only be used under `targets[_].code[_].[engine: Rego].source` with `version: "v1"`. You do not need to `import rego.v1` to use rego v1 syntax.

Reviewer: ```suggestion
Rego v1 syntax can only be used under `targets[_].code[_].[engine: Rego].source` with `version: "v1"`. No need to add `import rego.v1` to use rego v1 syntax.
```
